src/transforms/safe/cleanup.js

```diff
@@ -101,6 +103,53 @@ function makeReturnsUndefined(fnBody) {
   return new Shift.FunctionBody({ directives: fnBody.directives, statements: fixed });
 }
 
+let booleanBinOps = ['==', '!=', '===', '!==', '<', '<=', '>', '>=', 'in', 'instanceof'];
+function returnsBoolean(expr) {
+  return expr.type === 'LiteralBooleanExpression'
+    || expr.type === 'UnaryExpression' && expr.operator === '!'
+    || expr.type === 'BinaryExpression' && booleanBinOps.includes(expr.operator)
+    || expr.type === 'BinaryExpression' && (expr.operator === '&&' || expr.operator === '||') && returnsBoolean(expr.left) && returnsBoolean(expr.right)
+    || expr.type === 'BinaryExpression' && expr.operator === ',' && returnsBoolean(expr.right)
+    || expr.type === 'ConditionalExpression' && returnsBoolean(expr.consequent) && returnsBoolean(expr.alternate);
+}
+
+/*
+This function takes an expression like
+
+  a && (b, c, d) && e
```

Comment:
You have an example input but then no example of the output given this input?

Reply:
Updated.